test/test_server.py

```diff
@@ -77,9 +77,10 @@ class Utils(unittest.TestCase):
     def testMimeTypes(self):
         self.assertEqual(get_file_mime_type('a.html'), 'text/html')
         self.assertEqual(get_file_mime_type('a.gif'), 'image/gif')
+        self.assertEqual(get_file_mime_type('a.gif.jpg'), 'image/jpeg')
 
     def testMimeTypesUnknown(self):
-        runs = ['', '.', 'bbb', 'bbb.bbbb', '/', ' ']
+        runs = ['', '.', 'bbb', 'bb.', 'bbb.bbbb', '/', ' ']
```

Comment:
Since you've removed `.` from extension it is totally make sense to add "almost" valid unittests like
`jpg`